torch/_inductor/kernel/flex_attention.py

```diff
@@ -613,20 +628,20 @@ def flex_attention(*args, **kwargs):
 
 
 def flex_attention_backward_grid(
-    batch_size, num_heads, num_queries, d_model, num_key_value, meta
+    batch_size, q_heads, num_queries, d_model, kv_heads, num_key_value, meta
 ):
     """How is this kernel parallelized?
-    Currently this is only parallelizing over batch * num_heads, but we can, and want to
+    Currently this is only parallelizing over batch* kv_heads, but we can, and want to
     parallelize over ceil_div(num_key_value, key_value_block_size). To do this will either require
     atomic updates to some grad values or to have a two pass kernel design.
     """
     import triton
 
     return (
-        triton.cdiv(num_queries, meta["BLOCK_M2"])
+        triton.cdiv(num_queries, meta["BLOCK_M2"]) * (q_heads // kv_heads)
         + triton.cdiv(num_key_value, meta["BLOCK_N1"]),
```

Comment:
Also, it seems that we're generally launching *less* blocks here? It might be cleaner to just split it up over the `kv`heads as well, and then launch a separate clean-up reduction.